blueocean-dashboard/src/main/js/creation/git/ConnectStep.jsx

```diff
@@ -0,0 +1,190 @@
+/**
+ * Created by cmeyers on 10/19/16.
+ */
+import React, { PropTypes } from 'react';
+import FlowStep from '../FlowStep';
+
+const CREDENTIAL_CHOICE = {
+    SSH: 'SSH',
+    SYSTEM_SSH: 'SYSTEM_SSH',
+    USER_PASS: 'USER_PASS',
+};
+
+/**
+ * Component that accepts repository URL and credentials to initiate
+ * creation of a new pipeline.
+ */
+export default class ConnectStep extends React.Component {
```

Comment:
Renamed in bd2f370 - good suggestion.

Right, the intent is that everything in the "git" subfolder could be picked up and dropped into its own plugin. The code does use the `MultiStepFlow` and `FlowStep` so we either need to make that available via DI (Keith's work) or put in blueocean-core-js. Right now the `GitScmProvider` is registering itself via the extension point, so this is all built in a way that plugins can implement their own creation flows.